### PR TITLE
Fix lein eastwood warnings and some typos

### PR DESCRIPTION
**Pull request by laurio**

### CODE REVIEW

File: src/digest.clj [hunk 1]
```diff
@@ -1,10 +1,8 @@
 (ns digest
-  #^{ :author "Miki Tebeka <miki.tebeka@gmail.com>"
```
Comment (tebeka) [inline src/digest.clj]: Why remove the meta data (OK for my name, but the doc?)

Comment (laurio) [reply src/digest.clj]: Oh, I'm sorry about removing your name and doc from require - i executed `lein slamhound` in project folder which optimised the require but also removed the meta data. Will restore it!

Comment (laurio) [reply src/digest.clj]: `(:require .. :refer ..)` is a new way to do the same thing that allows effectively deprecate `:use`, which has some drawbacks. Please see https://stackoverflow.com/questions/871997/difference-between-use-and-require#comment25370819_16429572 for more details.

File: src/digest.clj [hunk 2]
```diff
@@ -71,29 +70,30 @@
...
   [algorithm-name]
   (let [update-meta (fn [meta]
                       (assoc meta
-                             :doc (str "Encode the given message with the " algorithm-name " algorithm.")
-                             :arglists '([message])))]
-    (-> (intern 'digest
-                (symbol (lower-case algorithm-name))
+                        :doc (str "Encode the given message with the " algorithm-name " algorithm.")
+                        :arglists '([message])))]
+    (-> 'digest
```
[the remark this answers is not in the document]
Comment (laurio) [reply src/digest.clj]: I modified threading in order to get rid of `lein kibit` warning - will restore the original version.

File: src/digest.clj [hunk 3]
```diff
@@ -71,29 +70,30 @@
...
         (alter-meta! update-meta))))
 
-(defn- create-fns []
+(defn- create-fns
   "Create utility function for each digest algorithms.
    For example will create an md5 function for MD5 algorithm."
-  (doseq [algorithm (algorithms)]
-    (create-fn! algorithm)))
+  []
+  (run! create-fn! (algorithms)))
```
[the remark this answers is not in the document]
Comment (laurio) [reply src/digest.clj]: The only difference between `run!` and `doseq` seems to be that `run!` doesn't support bindings which makes the code a bit shorter. Will restore `doseq`.

**Pull request merged**